packages/components/src/DateTimePickers/pickers/TimePicker/TimePicker.component.js

```diff
@@ -84,6 +84,7 @@ export class TimePicker extends React.Component {
 		this.props.onSubmit(event, {
 			textInput: option.label,
 			time: option.value,
+			origin: 'PICKER',
```

Comment:
Can you set that in `InputTimePicker` please ?
(And it's a little out of this PR scope, but could you rename the onSubmit name to onChange ?)